webgrid/renderers.py

```diff
@@ -1219,166 +1214,151 @@ def get_add_filter_row(self):
-
-    @property
-    def name(self):
-        return 'xls'
-
-    def __init__(self, grid, max_col_width=150):
-        warnings.warn(
-            "XLS support is deprecated and will be removed in a future version",
-            DeprecationWarning
-        )
-        super().__init__(grid)
-        self.define_styles()
-        self.col_contents_widths = defaultdict(int)
-        self.max_col_width = max_col_width
-
-    def render(self):
-        return self.build_sheet()
-
-    def can_render(self):
-        total_rows = self.grid.record_count + 1
-        if self.grid.subtotals != 'none':
-            total_rows += 1
-        return total_rows <= 65536 and sum(1 for _ in self.columns) <= 256
-
-    def define_styles(self):
-        self.style = BlankObject()
-        self.style.bold = xlwt.easyxf('font: bold True;')
-
-    def sanitize_sheet_name(self, sheet_name):
-        return sheet_name if len(sheet_name) <= 30 else (sheet_name[:27] + '...')
+def xlsxwriter_workbook_factory(*args, **kwargs):
+    '''
+    Wraps the standard xlsxwriter workbook class with an "adapter" layer to aid in dealing with
+    either openpyxl or xlsxwriter
+    '''
+    class XLSXWriterWorkbook(xlsxwriter.Workbook):
+        def __init__(self, *args, **kwargs):
+            self.use_openpyxl = False
+            super().__init__(*args, **kwargs)
+
+        def merged_totals_cell(self, xlh, value, style, colspan):
+            xlh.ws.merge_range(
+                xlh.rownum,
+                xlh.colnum,
+                xlh.rownum,
+                xlh.colnum + colspan - 1,
+                value,
+                style
+            )
 
-    def build_sheet(self, wb=None, sheet_name=None):
-        if xlwt is None:
-            # !!!: translate?
-            raise ImportError('you must have xlwt installed to use Excel renderer')
+        def merged_heading_cell(self, xlh, value, style, col_index, colspan):
+            xlh.ws.merge_range(
+                0, col_index, 0, col_index + (colspan - 1),
+                value,
+                style
+            )
 
-        if not self.can_render():
-            raise RenderLimitExceeded('Unable to render XLS sheet')
+    if xlsxwriter:
+        return XLSXWriterWorkbook(*args, **kwargs)
 
-        if wb is None:
-            wb = xlwt.Workbook()
-        sheet = wb.add_sheet(
-            self.sanitize_sheet_name(sheet_name or self.grid.ident)
-        )
-        xlh = Writer(sheet)
 
-        self.sheet_header(xlh)
-        self.sheet_body(xlh)
-        self.sheet_footer(xlh)
-        self.adjust_col_widths(sheet)
+class WorkbookBase:
+    """
+    Supports both xlsxwriter and openpyxl but matching the API of xlsxwriter to avoid
+    breaking changes.
 
-        return wb
+    Dispatches a workbook class depending on what libraries are available.
+    """
 
-    def sheet_header(self, xlh):
-        pass
+    def __new__(cls, *args, **kwargs):
+        if openpyxl and openpyxl.__class__.__name__ == 'module':
```

Comment:
I think at this point we'd want xlsxwriter usage relegated to backwards-compatibility. I think blazeutils defaults to using openpyxl if it is available. We could do the same here for now, but we'd need a clear way for a dev to manually make the renderer use xlsxwriter if they have a report already written that uses it.